In [1]:
import pandas as pd

In [5]:
df = pd.read_csv('criminal_main.csv',encoding='latin-1')

In [6]:
df.head()

Unnamed: 0                File  CaseName CivilKriminal         Court  \
0           0  2009_08295.htm.txt       NaN             K  County Court   
1           1  2009_08296.htm.txt       NaN             K  County Court   
2           2  2009_08297.htm.txt       NaN             K  County Court   
3           3  2009_08298.htm.txt       NaN             K  County Court   
4           4  2009_08311.htm.txt       NaN             K  County Court   

            County                        Judge        DistrictAttorney  \
0   Niagara County  Peter L. Broderick, Sr., J.     Michael J. Violante   
1    Oneida County         Michael L. Dwyer, J.       Scott D. McNamara   
2  Onondaga County          Joseph E. Fahey, J.  William J. Fitzpatrick   
3   Niagara County    Angelo J. Morinello, A.J.                     NaN   
4  Onondaga County          Joseph E. Fahey, J.  William J. Fitzpatrick   

                 ADA                                Keywords ...  \
0   Thomas H. Brandt                                affirmed ...   
1      Steven G. Cox            affirmed;interest of justice ...   
2  Victoria M. White   affirmed;modified;interest of justice ...   
3                NaN                                affirmed ...   
4   James P. Maxwell  affirmed;dismissed;interest of justice ...   

             Defense  First_Date Appeal_Date FirstD_dow  AppealD_dow  \
0                NaN     11/9/07    11/13/09     Friday       Friday   
1                NaN     11/6/06    11/13/09     Monday       Friday   
2  Legal Aid Society    10/18/06    11/13/09  Wednesday       Friday   
3                NaN     11/2/07    11/13/09     Friday       Friday   
4  Legal Aid Society     4/21/05    11/13/09   Thursday       Friday   

  FirstD_weekday AppealD_weekday FirstD_month AppealD_month  Y  
0              1               1         11.0            11  0  
1              1               1         11.0            11  0  
2              1               1         10.0            11  1  
3              1               1         11.0            11  0  
4              1               1          4.0            11  1  

[5 rows x 28 columns]

#### Part I. GroundsForAppeal column

Assuming that there was more than merely harmless error, there are four basic grounds for appeal:

- The lower court made a serious error of law (plain error);
- The weight of the evidence does not support the verdict;
- The lower court abused its discretion in making an errant ruling;
- The claim of Ineffective Assistance of Counsel under the Sixth Amendment.

In [55]:
grounds = df.GroundsForAppeal

In [56]:
num_rows = grounds.shape[0]

In [57]:
num_missing = grounds.isnull().sum()

In [58]:
share_missing = num_missing/num_rows
print('Share of missing values:', str(share_missing))

Share of missing values: 0.472076884948


In [59]:
unique  = grounds.unique()

In [60]:
grounds_list = []

for line in grounds:
    if isinstance(line, float): #skip nan
        continue
    words = line.split(';')
    grounds_list.extend(words)

In [64]:
unique_grounds = set(grounds_list)

In [65]:
unique_grounds

{'coerce',
 'coercion',
 'double jeopardy',
 'incapac',
 'ineffective counsel',
 'instructions',
 'juror',
 'mental',
 'resentenc',
 'sever',
 'speedy',
 'sufficient',
 'suppress',
 'youth'}

In [66]:
#Count number of unique occurrences per category - this does
for ground in unique_grounds:
    print(ground, grounds_list.count(ground))

instructions 1556
suppress 5042
mental 3119
coercion 365
ineffective counsel 3
youth 740
sufficient 9569
coerce 552
sever 5128
double jeopardy 382
speedy 500
resentenc 3096
incapac 153
juror 2038


In [67]:
len(grounds)

36314

#### Conclusion: about half of the data in the column is missing. However, we could clean and featurize existing categories.

#### Part II. Mode of Conviction column

In [40]:
mode = df.ModeOfConviction

In [34]:
unique_modes = mode.unique()

In [35]:
num_rows = mode.shape[0]
num_missing = mode.isnull().sum()

In [36]:
share_missing_mode = num_missing/num_rows
print("Share of missing mode of conviction:",str(share_missing_mode))

Share of missing mode of conviction: 0.461998127444


In [52]:
#Count number of unique occurrences per category
mode_list = mode.tolist()
for ex_mode in unique_modes:
    if isinstance(ex_mode,float):
        continue #ignore nan
    print(ex_mode, mode_list.count(ex_mode))

jury verdict 7852
plea of guilty 10690
nonjury trial 995


#### Conclusion: only 46% of values are present. Could featurize if needed